### PR TITLE
Update SOR to let v2-sdk handle the FOT calculation

### PR DESCRIPTION
**Pull request by mikeki**
# Let V2-SDK handle the calculation for FOT

There is a bug in how we calculate FOT tax, but it is now fixed in v2-sdk and this should make this code cleaner

### CODE REVIEW

File: src/providers/v2/quote-provider.ts [hunk 1]
```diff
@@ -157,54 +103,11 @@ export class V2QuoteProvider implements IV2QuoteProvider {
...
-                    inputAmount.quotient
-                  );
-                  [inputAmount] = pair.getInputAmount(inputAmountWithBuyFeeBps);
-                } else {
-                  [inputAmount] = pair.getInputAmount(inputAmount);
-                }
-              } else {
-                [inputAmount] = pair.getInputAmount(inputAmount);
-              }
+            for (const pair of route.pairs.reverse()) {
```
Comment (mikeki) [inline src/providers/v2/quote-provider.ts:106]: Please help me double check that this for loop remains the same

File: src/providers/v2/quote-provider.ts [hunk 2]
```diff
@@ -92,62 +90,10 @@ export class V2QuoteProvider implements IV2QuoteProvider {
...
-                  const [outputAmountNew] = pair.getOutputAmount(outputAmount);
-                  outputAmount = outputAmountNew;
-                }
-              } else {
-                const [outputAmountNew] = pair.getOutputAmount(outputAmount);
-                outputAmount = outputAmountNew;
-              }
+              [outputAmount] = pair.getOutputAmount(
+                outputAmount,
+                providerConfig.enableFeeOnTransferFeeFetching ?? false
```
Comment (jsy1218) [inline src/providers/v2/quote-provider.ts:95]: nit - this reads a little weird. `providerConfig.enableFeeOnTransferFeeFetching === true` seems more direct.

File: src/providers/v2/quote-provider.ts (hunk 1 above)
Comment (jsy1218) [inline src/providers/v2/quote-provider.ts:106]: why is there a reverse?

Comment (jsy1218) [reply src/providers/v2/quote-provider.ts:106]: aw I see at this point, prefer to retain the same code:

```
for (let i = route.pairs.length - 1; i >= 0; i--) {
              const pair = route.pairs[i]!;
}
```

**Review by jsy1218 — APPROVED** [review]

**Review by jsy1218 — APPROVED** [review]

File: src/providers/v2/quote-provider.ts (hunk 1 above)
Comment (jsy1218) [reply src/providers/v2/quote-provider.ts]: Looks fine!

**Pull request merged**